DeepLabCut/DeepLabCut
# Add build matrix for docker images, update images

- Improvements to the build process for docker images: With the specified build matrix, it is now easier to build images with different DeepLabCut and CUDA base container versions
- Support two version of deeplabcut, as well as CUDA 11.4 (as before) and the more recent CUDA 11.7

Open:
- [x] Build containers, and attach the full build log
- [x] Update containers on DockerHub
- [x] Discuss addition of additional CUDA versions/base images
- [ ] Update PyPI package to interface with the containers

## Build and test logs

https://gist.github.com/stes/80412bac41a37cabb6248db5e5d2884e

## Testing

If you come across this PR and want to test it, please see [this discussion](https://forum.image.sc/t/deeplabcut-docker-tensorflow-gpu-problem/70044/7?u=stes) for instructions (until this package is updated):

- Download [this experimental version of `deeplabcut-docker`](https://gist.github.com/stes/4dfb0c7e95a3ac3b2356551e8d375254)
- Run the following:
```
# make executable
chmod +x ./deeplabcut_docker.sh
# pull the image
docker pull deeplabcut/deeplabcut:2.2.1.1-gui-cuda11.7.0-runtime-ubuntu20.04
# run the UI
./deeplabcut_docker.sh gui
```

This should open the deeplabcut UI. If you are on a linux machine, you might run `xhost +` before executing the last command if you get a permission error.

Fixes #2164
